File: uikit/src/main/java/com/sendbird/uikit/activities/viewholder/OpenChannelAdminMessageViewHolder.java
package com.sendbird.uikit.activities.viewholder;

import android.view.View;

import androidx.annotation.NonNull;

import com.sendbird.android.channel.BaseChannel;
import com.sendbird.android.message.BaseMessage;
import com.sendbird.uikit.consts.MessageGroupType;
import com.sendbird.uikit.databinding.SbViewOpenChannelAdminMessageBinding;
import com.sendbird.uikit.widgets.OpenChannelAdminMessageView;

import java.util.Map;

public final class OpenChannelAdminMessageViewHolder extends MessageViewHolder {
    @NonNull
    private final OpenChannelAdminMessageView openChannelAdminMessageView;

    OpenChannelAdminMessageViewHolder(@NonNull SbViewOpenChannelAdminMessageBinding binding, boolean useMessageGroupUI) {
        super(binding.getRoot(), useMessageGroupUI);
        openChannelAdminMessageView = binding.openChannelAdminMessageView;
    }

    @Override
    public void bind(@NonNull BaseChannel channel, @NonNull BaseMessage message, @NonNull MessageGroupType messageGroupType) {
        openChannelAdminMessageView.drawMessage(message);
    }

    @NonNull
    @Override
    public Map<String, View> getClickableViewMap() {
        return clickableViewMap;
    }
}

File: uikit/src/main/java/com/sendbird/uikit/internal/ui/viewholders/OpenChannelAdminMessageViewHolder.kt
package com.sendbird.uikit.internal.ui.viewholders

import android.view.View
import com.sendbird.android.channel.BaseChannel
import com.sendbird.android.message.BaseMessage
import com.sendbird.uikit.activities.viewholder.MessageViewHolder
import com.sendbird.uikit.databinding.SbViewOpenChannelAdminMessageBinding
import com.sendbird.uikit.model.MessageListUIParams

internal class OpenChannelAdminMessageViewHolder internal constructor(
    val binding: SbViewOpenChannelAdminMessageBinding,
    messageListUIParams: MessageListUIParams
) : MessageViewHolder(binding.root, messageListUIParams) {

    override fun bind(channel: BaseChannel, message: BaseMessage, messageListUIParams: MessageListUIParams) {
        binding.openChannelAdminMessageView.messageUIConfig = messageUIConfig
        binding.openChannelAdminMessageView.drawMessage(message)
    }

    override fun getClickableViewMap(): Map<String, View> = mapOf()
}
